pkg/deploy/generator/resources_gateway.go

```diff
@@ -318,7 +318,11 @@ EOF
 
 export AZURE_CLOUD_NAME=$AZURECLOUDNAME
 az login -i --allow-no-subscriptions
-az account set -s "$SUBSCRIPTIONID"
+
+# Running "az account set" is not working in the pipeline.
+# we are running VM ScaleSet in a named ResourceGroup, it should
+# be OK to to remove this set subscription
+# az account set -s "$SUBSCRIPTIONID"
```

Comment:
Why is it failing?

> The subscription of 'blah' doesn't exist in cloud 'AzureCloud'.

I don't get it: `pkg/deploy/generator/resources_rp.go` has exactly the same piece. Why we did not have an issue there?

Reply:
We don't know.  This was introduced for the RP earlier this year.  Gateway deployment to eastus2euap was fine without this line and now it is also failing with this as well.
```
++ export AZURE_CLOUD_NAME=AzureCloud
++ AZURE_CLOUD_NAME=AzureCloud
++ az login -i --allow-no-subscriptions
++ az account set -s my-awesome-subscription-id
ERROR: The subscription of 'my-awesome-subscription-id' doesn't exist in cloud 'AzureCloud'.
``` 

Where the subscription ID is the canary sub id.  Perhaps there's something in the RG setup that is missed.  Regardless, this isn't a necessary step as this code is running on the VM's in a VMSS deployed to a RG which is in the subscription.